pypeit/spectrographs/keck_esi.py

```diff
@@ -0,0 +1,404 @@
+            to 1 and an unmasked value set to 0.  All values are set to
+            0.
+        """
+        # Call the base-class method to generate the empty bpm
+        bpm_img = super().bpm(filename, det, shape=shape, msbias=msbias)
+
+
+        # Get the binning 
+        msgs.info("Custom bad pixel mask for ESI")
+        hdu = io.fits_open(filename)
+        binspatial, binspec = parse.parse_binning(hdu[0].header['BINNING'])
+        hdu.close()
+
+        # Binning Independent Masking 
+        bpm_img[:,0:(2//binspatial)] = 1
+        # Mask out the 'hotspot' on the upper left coner 
+        bpm_img[(3842//binspec):(3944//binspec), (19//binspatial):(161//binspatial)] = 1
+        # Mask out the 'bad columns' on the upper left coner
+        bpm_img[(2642//binspec):, (418//binspatial):(442//binspatial)] = 1
+
+        # Return
+        return bpm_img
+
+    @property
+    def norders(self):
+        """
+        Number of orders for this spectograph. Should only defined for
+        echelle spectrographs, and it is undefined for the base class.
+        """
+        return 10   # 15-6
+
+    @property
+    def order_spat_pos(self):
+        """
+        Return the expected spatial position of each echelle order.
+        """
+        return np.array([0.115, 0.245, 0.362, 0.465, 0.558, 0.642, 0.719, 0.791, 0.861, 0.933])
+
+    @property
+    def order_spat_width(self):
+        """
+        Return the expected spatial width of each slit trace for each order,
+        relative to the spatial size of the detector.
+        """
+        return np.array([0.0879, 0.0818, 0.0779, 0.0747, 0.0720, 0.0696, 0.0676, 0.0658, 0.0640,
+                         0.0617])
+
+    @property
+    def orders(self):
+        """
+        Return the order number for each echelle order.
+        """
+        return np.arange(15, 5, -1, dtype=int)
+
+    @property
+    def spec_min_max(self):
+        """
+        Return the minimum and maximum spectral pixel expected for the
+        spectral range of each order.
+        """
+        spec_max = np.full(self.norders, np.inf)
+        spec_min = np.full(self.norders, -np.inf)
+        return np.vstack((spec_min, spec_max))
+
+    def order_platescale(self, order_vec, binning=None):
+        """
+        Return the platescale for each echelle order.
+
+        This routine is only defined for echelle spectrographs, and it is
+        undefined in the base class.
+
+        Args:
+            order_vec (`numpy.ndarray`_):
+                The vector providing the order numbers.
+            binning (:obj:`str`, optional):
+                The string defining the spectral and spatial binning.
+
+        Returns:
+            `numpy.ndarray`_: An array with the platescale for each order
+            provided by ``order``.
+        """
+        norders = len(order_vec)
+        binspatial, binspec = parse.parse_binning(binning)
+        return np.full(norders, 0.30*binspatial)
```

Comment:
should this be: `return np.full(norders, 0.1542*binspatial)`, where `0.1542` is the platescale for ESI?

Reply:
whoa, good catch!